hotsos/defs/scenarios/kernel/memory.yaml

```diff
@@ -44,4 +58,23 @@ conclusions:
         for more info.
       format-dict:
         pcent: $compaction_failures_percent
-
+  too_many_free_hugepages:
+    decision: too_many_free_hugepages
+    raises:
+      type: MemoryWarning
+      message: >-
+        This system is using huge pages which are consuming {hugetlb} GB, that is {hugepcent}% out
+        of {memtotal} GB total system memory. Available memory is only {memavail} GB, which is
+        less than {availthreshold}%. At the same time only {hpusedpcent}% of the huge pages are being
+        actively used. Memory reserved for the huges pages can't be used for any other purpose but
+        the system maybe be suffering from having not enough available memory. Review /proc/meminfo,
+        your current huge pages configuration and check if you can allocate less memory for the
+        huge pages. For details, see the kernel documentation at
+        https://docs.kernel.org/admin-guide/mm/hugetlbpage.html.
```

Comment:
I'd like to propose reworking this slightly as follows:

```
This host is using hugepages which are consuming {hugetlb} out of {memtotal}GB total system memory i.e. {hugepcent}%. This leave {memavail}GB available memory for processes not using hugepages. Also, only {hpusedpcent}% of the hugepages are being actively used. Memory reserved for hugepages can't be used for any other purpose which may starve the system of available memory. Please review /proc/meminfo and your current hugepages configuration to see if you can reduce the hugepages allocation. For details, see the kernel documentation at https://docs.kernel.org/admin-guide/mm/hugetlbpage.html.
```

Reply:
done